lib/rails/dom/testing/assertions/selector_assertions/html_selector.rb

```diff
@@ -74,8 +74,7 @@ def extract_selector
       raise ArgumentError, "Expecting a selector as the first argument"
     end
 
-    context.substitute!(selector, @values)
-    selector
+    @selector = context.substitute!(selector, @values)
```

Comment:
The `!` indicates `substitute!` mutates the argument, so why don't we put the `dup` in here?